### DISCUSSION TITLE
Please help me

### DISCUSSION THREAD

**Discussion by LavenderZZZ** [relab/hotstuff#42] [General]
Hello !     I am sorry to bother you . We just followed the steps to get started your project and simply executed ./hotstuff run , but it turned out that either the "command not accepted" or "Block not voted for".  We are beginners to block chain, I wonder if you can give us some suggestions about how to make the propose accepted.

       Plus, if we wanna use your framework to develop an application ,could you please tell us  whether it is possible to embed your codes into our application which can provide a platform where we could upload data to the block chain and search data from that using the hotstuff consensus provided.

![image](https://user-images.githubusercontent.com/72267499/160276920-b552e33d-ca89-402f-a1f5-110a80221126.png)

**Comment by johningve** [comment]
Hi!

None of these log messages indicate any serious issues. It's likely that they were all caused by messages arriving in a different order than expected. That can happen often, especially in a low-latency network.

If you take a look at the final log line from the client, you will see that all of the commands it sent to the replicas were executed successfully, with no failures or timeouts.

**Comment by johningve** [comment]
I don't think it will be too hard to embed the framework in another application. You should be able to use the `consensus` package with its modules at least. You can, and probably will need to, write your own blockchain module in order to add the features you mentioned.
